### DISCUSSION TITLE
Possible Wallet Mounting Error

### DISCUSSION THREAD

**Discussion by Tarranzilla** [mercadopago/sdk-react#99] [Q&A] [answered]
Hello guys, I am a new developer - so sorry if this mistake might look dumb.

I am using Next JS and I am mounting the Wallet Component, at my checkout page, but it seems it is being mounted twice, and sometimes more than twice.

And I think this error occurs because the wallet is not able to unmount some DOM element. I also have a GET 405 error when mounting it.

I was creating the preference before going to the checkout page, but due to my logic - I changed to creating the preference when clicking the button - so now I also get this get_preference_details_failed. ( But i guess it is not critical right? kkkk )

Other than that the wallet works fine, I am able to process the payment and make tests - But i am getting this error and it is really annoying and breaking my website experience. I cannot find resources to help me at mercado pago dev docs.

Those are the errors i get: ( i replaced my public key to not show it here )

`GET https://api.mercadopago.com/bricks/preferences/?public_key=MY_TEST_KEY&locale=en&js_version=2.0.0&referer=http%3A%2F%2Flocalhost%3A667 405 (Method Not Allowed)`

`{type: 'non_critical', cause: 'get_preference_details_failed', message: 'Failed to get preference details'}
cause
: 
"get_preference_details_failed"
message
: 
"Failed to get preference details"
type
: 
"non_critical"`

`Uncaught (in promise) TypeError: Failed to execute 'removeChild' on 'Node': parameter 1 is not of type 'Node'.
    at be.<anonymous> (<anonymous>:1:127468)
    at Generator.next (<anonymous>)
    at a (<anonymous>:1:125383)`
    
  And this is how part of my code related to the wallet looks:
  
  `useEffect(() => {
        // Initialize Mercado Pago SDK
        if (process.env.NEXT_PUBLIC_MERCADO_PAGO_PUBLIC_KEY !== undefined) {
            initMercadoPago(process.env.NEXT_PUBLIC_MERCADO_PAGO_PUBLIC_KEY);
        }
    }, [])`
  
  `<div id="walletBrick_container" className={cartItems.length > 0 ? "Wallet" : "Wallet Disabled"}>
                                    <Wallet
                                        initialization={{ redirectMode: "modal" }}
                                        customization={{ texts: { action: "buy", valueProp: "smart_option" } }}
                                        onSubmit={handleMercadoClick}
                                        onReady={() => {
                                            setMpWalletLoaded(true);
                                            console.log("Wallet ready");
                                        }}
                                        locale="pt-BR"
                                    />
                                </div>`

What should i do? I would appreciate the help!

**Comment by lucmkz** [comment, marked as answer]
Good afternoon @Tarranzilla , how are you?

First of all, no question is dumb, and it is always welcome!

It's good to know that the component works well, regarding multiple renderings and the difficulty of handling using next js, I'll refer you to this [wiki link](https://github.com/mercadopago/sdk-react/wiki/Using-Bricks-with-Next.js), here are some tips on how to use this SDK in next js.

I also believe that it could be a failure in the application life cycle and is triggering some renderings at inappropriate times. I will indicate a few good reading links that can help you in dealing with this type of problem.

[React.dev - Memo](https://react.dev/reference/react/memo)
[Medium - React Components rendered twice](https://andreasheissenberger.medium.com/react-components-render-twice-any-way-to-fix-this-91cf23961625)
[Medium - Wait, you’re not using <React.StrictMode>?!](https://medium.com/nmc-techblog/wait-youre-not-using-react-strictmode-a9713927a33b)

If none of these articles help you, I hope you add more snippets from your complete implementation.
Thank you and see you soon!

**Reply by Tarranzilla** [reply]
Hello Lucas, i followed the wiki link and built a similar approach for using the Wallet component. It works now, and doesnt render twice since i am importing it dynamically - Thank you!

But now i am facing another challenge! I am using the modal redirect mode, and generating the preference on submit, it all works fine until i submit the Test Credit Card information. After i submit it i get an error: FL-0eb55368

I dont know what is happening, but in my console i can see some errors, i dont know if they are related to this issue:

[Report Only] Refused to load the script 'https://http2.mlstatic.com/analytics/ga/mlb-mp-analytics.min.js' because it violates the following Content Security Policy directive: "script-src 'nonce-EBDFn+zUPkH3d1Xb+70+Qg==' 'strict-dynamic' 'unsafe-eval' 'report-sample' https: 'unsafe-inline'". Note that 'strict-dynamic' is present, so host-based allowlisting is disabled. Note that 'script-src-elem' was not explicitly set, so 'script-src' is used as a fallback.
Understand this error
modal/?preference-id=201803820-13921b00-9a94-4c9d-9bef-8783289c80d4&from-widget=true&sniffing-rollout=sniffing-api&router-request-id=6a1c3a4d-944b-49c6-bca8-f4a7734725b2:5 
        
        
       GET https://http2.mlstatic.com/frontend-assets/px-checkout-frontend/undefined net::ERR_ABORTED 404 (Not Found)
nrWrapper @ modal/?preference-id=201803820-13921b00-9a94-4c9d-9bef-8783289c80d4&from-widget=true&sniffing-rollout=sniffing-api&router-request-id=6a1c3a4d-944b-49c6-bca8-f4a7734725b2:5
task @ modal/?preference-id=201803820-13921b00-9a94-4c9d-9bef-8783289c80d4&from-widget=true&sniffing-rollout=sniffing-api&router-request-id=6a1c3a4d-944b-49c6-bca8-f4a7734725b2:315
postTask (async)
enqueueTask @ modal/?preference-id=201803820-13921b00-9a94-4c9d-9bef-8783289c80d4&from-widget=true&sniffing-rollout=sniffing-api&router-request-id=6a1c3a4d-944b-49c6-bca8-f4a7734725b2:262
loadScripts @ modal/?preference-id=201803820-13921b00-9a94-4c9d-9bef-8783289c80d4&from-widget=true&sniffing-rollout=sniffing-api&router-request-id=6a1c3a4d-944b-49c6-bca8-f4a7734725b2:315
onLoad @ modal/?preference-id=201803820-13921b00-9a94-4c9d-9bef-8783289c80d4&from-widget=true&sniffing-rollout=sniffing-api&router-request-id=6a1c3a4d-944b-49c6-bca8-f4a7734725b2:412
nrWrapper @ modal/?preference-id=201803820-13921b00-9a94-4c9d-9bef-8783289c80d4&from-widget=true&sniffing-rollout=sniffing-api&router-request-id=6a1c3a4d-944b-49c6-bca8-f4a7734725b2:5
load (async)
nrWrapper @ modal/?preference-id=201803820-13921b00-9a94-4c9d-9bef-8783289c80d4&from-widget=true&sniffing-rollout=sniffing-api&router-request-id=6a1c3a4d-944b-49c6-bca8-f4a7734725b2:5
(anonymous) @ modal/?preference-id=201803820-13921b00-9a94-4c9d-9bef-8783289c80d4&from-widget=true&sniffing-rollout=sniffing-api&router-request-id=6a1c3a4d-944b-49c6-bca8-f4a7734725b2:411
(anonymous) @ modal/?preference-id=201803820-13921b00-9a94-4c9d-9bef-8783289c80d4&from-widget=true&sniffing-rollout=sniffing-api&router-request-id=6a1c3a4d-944b-49c6-bca8-f4a7734725b2:415Understand this error
modal/:1 Refused to execute script from 'https://http2.mlstatic.com/frontend-assets/px-checkout-frontend/undefined' because its MIME type ('text/html') is not executable, and strict MIME type checking is enabled.

Maybe an undefined value is being passed to this px-checkout-frontend? but how can i control this?

Anyways, here is the link to my code if you want to have a look:

https://github.com/Tarranzilla/mercado-pago-bricks-with-firebase

![image](https://github.com/mercadopago/sdk-react/assets/54077894/2155dcdb-4ed7-478e-9cf7-ecbb91630ef6)

**Reply by Tarranzilla** [reply]
Já que vi que tu é Brasileiro vou falar em português mesmo!

Andei lendo a comunidade do discord e parece que há algum problema com o init_point quando criamos a preferência, to tentando descobrir aqui se é isso. Buscando alguma maneira de configurar o init_point e o sandbox_init_point. 

Quando eu uso minhas credenciais de Produção o erro nao parece ocorrer, mas eu nao consigo testar o pagamento de teste pois acredito que nao seja possivel no modo de produão fazer isto...

Se houver alguma informação, agradeço!
